File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/card/AddCardUseCase.java
package com.lattis.ellipse.domain.interactor.card;

import com.lattis.ellipse.data.network.model.response.card.AddCardResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.CardRepository;

import org.json.JSONObject;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 7/26/17.
 */

public class AddCardUseCase extends UseCase<AddCardResponse> {

    private CardRepository cardRepository;
    private String cardNumber;
    private int expiry_month;
    private int expiry_year;
    private String cvc;
    private JSONObject intent;


    @Inject
    public AddCardUseCase(ThreadExecutor threadExecutor,
                             PostExecutionThread postExecutionThread,
                          CardRepository cardRepository) {
        super(threadExecutor, postExecutionThread);
        this.cardRepository = cardRepository;
    }

    public AddCardUseCase withCardNumber(String cardNumber) {
        this.cardNumber = cardNumber;
        return this;
    }

    public AddCardUseCase withExpiryMonth(int expiry_month) {
        this.expiry_month = expiry_month;
        return this;
    }

    public AddCardUseCase withExpiryYear(int expiry_year) {
        this.expiry_year = expiry_year;
        return this;
    }

    public AddCardUseCase withCVC(String cvc) {
        this.cvc = cvc;
        return this;
    }

    public AddCardUseCase withIntent(JSONObject intent) {
        this.intent = intent;
        return this;
    }

    @Override
    protected Observable<AddCardResponse> buildUseCaseObservable() {
        return cardRepository.addCard(cardNumber,expiry_month,expiry_year,cvc,intent);
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/card/AddCardUseCase.kt
package com.lattis.domain.usecase.card

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.CardRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import org.json.JSONObject
import javax.inject.Inject

class AddCardUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val cardRepository: CardRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var cardNumber: String? = null
    private var expiry_month = 0
    private var expiry_year = 0
    private var cvc: String? = null
    private var intent: JSONObject? = null
    fun withCardNumber(cardNumber: String?): AddCardUseCase {
        this.cardNumber = cardNumber
        return this
    }

    fun withExpiryMonth(expiry_month: Int): AddCardUseCase {
        this.expiry_month = expiry_month
        return this
    }

    fun withExpiryYear(expiry_year: Int): AddCardUseCase {
        this.expiry_year = expiry_year
        return this
    }

    fun withCVC(cvc: String): AddCardUseCase {
        this.cvc = cvc
        return this
    }

    fun withIntent(intent: JSONObject?): AddCardUseCase {
        this.intent = intent
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return cardRepository.addCard(cardNumber!!, expiry_month, expiry_year, cvc!!, intent!!)
    }

}